### ISSUE TITLE
[WIP] 4.0.0 Release Notes

### ISSUE THREAD

**Issue by antiochp** [mimblewimble/grin#3341]
### tl;dr

* New Cuckarooz PoW and header version 4 for latest hardfork.
* TUI optimizations and overal code improvements
* Server and backend DB code optimizations and improvements
* BlockHeader via API now includes all relevant fields (including MMR size)
* Feature flagged NRD kernel support (enabled in floonet, remains disabled in mainnet)

----

Highlights (non-exhaustive) from PRs included since `3.1.1` - 

* header version 4 accompanied by new AR PoW (https://github.com/mimblewimble/grin/pull/3334)
* Enable NRD kernel support (noop) feature flagged (https://github.com/mimblewimble/grin/pull/3303)
* Some optimizsations in TUI (https://github.com/mimblewimble/grin/pull/3325)
* Reduce memory allocations in PMMR (https://github.com/mimblewimble/grin/pull/3328)
* Adding kernel_mmr_size and output_mmr_size to BlockHeaderPrintable (https://github.com/mimblewimble/grin/pull/3329)
* Fix miminum share difficulty in Stratum Server (https://github.com/mimblewimble/grin/pull/3323)
* Always return a typed structure from lmdb store (https://github.com/mimblewimble/grin/pull/3312)
* Use generic types instead of trait objects in tx pool (https://github.com/mimblewimble/grin/pull/3308)
* Reduce number of allocations in Headers message read (https://github.com/mimblewimble/grin/pull/3301)
* Update Cursive to 0.14 and update backend (https://github.com/mimblewimble/grin/pull/3291)
* go back to storing header_head in the db (https://github.com/mimblewimble/grin/pull/3268)
* Use croaring-mw with 'compat' feature (https://github.com/mimblewimble/grin/pull/3274)
* Add better LMDB error output (https://github.com/mimblewimble/grin/pull/3263)

**Comment by wiesche89** [comment]
This was for the 4.0.0 release notes and looks obsolete now. I think we can close this old release tracking issue.
